File: app/src/main/java/com/bzh/dytt/data/network/DyttService.java
package com.bzh.dytt.data.network;

import android.arch.lifecycle.LiveData;

import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Url;

public interface DyttService {

    @GET("{detail_link}")
    Call<ResponseBody> getVideoDetail(@Path("detail_link") String detailLink);

    @GET("{category_string}")
    LiveData<ApiResponse<ResponseBody>> getMovieListByCategory(@Path("category_string") String category);

    @GET
    LiveData<ApiResponse<ResponseBody>> search(@Url String query);

    @GET
    Call<ResponseBody> getSearchVideoDetail(@Url String url);
}

File: app/src/main/java/com/bzh/dytt/data/network/DyttService.kt
package com.bzh.dytt.data.network

import android.arch.lifecycle.LiveData
import okhttp3.ResponseBody
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Url

interface DyttService {

    @GET("{detail_link}")
    fun getVideoDetail(@Path("detail_link") detailLink: String): Call<ResponseBody>

    @GET("{category_string}")
    fun getMovieListByCategory(@Path("category_string") category: String): LiveData<ApiResponse<ResponseBody>>

    @GET
    fun search(@Url query: String): LiveData<ApiResponse<ResponseBody>>

    @GET
    fun getSearchVideoDetail(@Url url: String): Call<ResponseBody>
}
